go/launcher/proxy/driverhub/debugger/debugger.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package debugger enables wTL Debugger.
+package debugger
+
+import (
+  "context"
+  "encoding/json"
+  "fmt"
+  "log"
+  "net"
+  "net/http"
+  "os"
+  "regexp"
+  "sync"
+
+  "github.com/bazelbuild/rules_webtesting/go/launcher/errors"
+)
+
+type breakpoint struct {
+  ID        int    `json:"id,omitempty"`
+  Path      string `json:"path,omitempty"`
+  pathRegex *regexp.Regexp
+  Methods   []string `json:"methods,omitempty"`
+}
+
+type command struct {
+  ID         int         `json:"id"`
+  Command    string      `json:"command,omitempty"`
+  Breakpoint *breakpoint `json:"breakpoint,omitempty"`
+}
+
+type request struct {
+  Method string `json:"method,omitempty"`
+  Path   string `json:"path,omitempty"`
+}
+
+type response struct {
+  ID      int     `json:"id,omitempty"`
```

Comment:
nit: Is there a reason why ID on the response is omitempty, by the ID on command is not?